src/adm/filter.rs

```diff
@@ -81,48 +97,42 @@ impl AdmFilter {
         let url = &tile.click_url;
         let species = "Click";
         tags.add_tag("type", species);
+        tags.add_extra("tile", &tile.name);
         tags.add_extra("url", &url);
 
-        // Check the required fields are present for the `click_url`
-        // pg 15 of 5.7.21 spec. (sort for efficiency)
-        // The list of sorted required query param keys for click_urls
-        let req_keys = vec!["aespFlag", "ci", "ctag", "key", "version"];
-        // the list of optionally appearing query param keys
-        let opt_keys = vec!["click-status"];
-
-        let mut all_keys = req_keys.clone();
-        all_keys.extend(opt_keys.clone());
-
+        // Check the required fields are present for the `click_url` pg 15 of
+        // 5.7.21 spec
         let parsed: Url = match url.parse() {
             Ok(v) => v,
             Err(e) => {
                 tags.add_extra("parse_error", &e.to_string());
                 return Err(HandlerErrorKind::InvalidHost(species, url.to_string()).into());
             }
         };
-        let mut query_keys = parsed
+        let query_keys = parsed
             .query_pairs()
             .map(|p| p.0.to_string())
-            .collect::<Vec<String>>();
-        query_keys.sort();
+            .collect::<HashSet<String>>();
```

Comment:
not that sets matter much here, mostly just killing the unneeded sort() and moving it to `lazy_static`